shopfloor_single_product_transfer/services/single_product_transfer.py

```diff
@@ -0,0 +1,588 @@
+        handlers = self._set_quantity__scan_product_handlers()
+        if product:
+            response = self._use_handlers(handlers, move_line, product)
+            if response:
+                return response
+
+    def _set_quantity__scan_lot(self, move_line, barcode, confirmation=False):
+        search = self._actions_for("search")
+        lot = search.lot_from_scan(barcode)
+        handlers = self._set_quantity__scan_product_handlers()
+        if lot:
+            product = lot.product_id
+            response = self._use_handlers(handlers, move_line, product, lot=lot)
+            if response:
+                return response
+
+    def _set_quantity__valid_dest_location_for_move_line_domain(self, move_line):
+        move_line_dest_location = move_line.location_dest_id
+        return [
+            "|",
+            ("id", "in", move_line_dest_location.ids),
+            ("id", "child_of", move_line_dest_location.ids),
+        ]
+
+    def _set_quantity__valid_dest_location_for_move_line(self, move_line):
+        domain = self._set_quantity__valid_dest_location_for_move_line_domain(move_line)
+        return self.env["stock.location"].search(domain)
+
+    def _valid_dest_location_for_menu_domain(self):
+        return [
+            "|",
+            ("id", "in", self.picking_types.default_location_dest_id.ids),
+            ("id", "child_of", self.picking_types.default_location_dest_id.ids),
+        ]
+
+    def _valid_dest_location_for_menu(self):
+        domain = self._valid_dest_location_for_menu_domain()
+        return self.env["stock.location"].search(domain)
+
+    def _set_quantity__check_location(self, location, move_line, confirmation=False):
+        valid_locations_for_move_line = (
+            self._set_quantity__valid_dest_location_for_move_line(move_line)
+        )
+        valid_locations_for_menu = self._valid_dest_location_for_menu()
+        message = False
+        if location in valid_locations_for_move_line:
+            # scanned location is valid, return no response
+            ...
+        elif location in valid_locations_for_menu:
+            # Considered valid if scan confirmed
+            if not confirmation:
+                # Ask for confirmation
+                orig_location = move_line.location_dest_id
+                message = self.msg_store.confirm_location_changed(
+                    orig_location, location
+                )
+            ...
+        else:
+            # Invalid location, return an error
+            message = self.msg_store.dest_location_not_allowed()
+        if message:
+            return self._response_for_set_quantity(move_line, message=message)
+
+    def _set_quantity__post_move(self, location, move_line, confirmation=False):
+        self._actions_for("stock")
+        # TODO qty_done = 0: transfer_no_qty_done
+        # TODO qty done < product_qty: transfer_confirm_done
+        picking = move_line.picking_id
+        picking._action_done()
```

Comment:
High: Before posting, the scanned location must be set on the move line

Reply:
`_write_destination_on_lines` is implemented in:
- `location_content_transfer`
- `zone_picking`
- `cluster_picking` (but it is called `_unload_write_destination_on_lines`)

And all of them has a different implementation... so for now I'll do the same: create our own method in the current scenario, add the support in `shopfloor_checkout_sync`. Refactoring this has to be planned, I prefer to not pass time on this right now.

Reply:
@jbaudoux would we need the `_lock_lines` method as well in this scenario? It is used in location content transfer, zone picking, cluster picking...

Reply:
seems safer indeed